modules/json/filterx-cached-json-file.c

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2024 Attila Szakacs
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "filterx-cached-json-file.h"
+#include "filterx/object-json.h"
+
+typedef struct FilterXFunctionCachedJsonFile_
+{
+  FilterXFunction super;
+  FilterXObject *cached_json;
```

Comment:
This needs to be frozen, as it is accessed by multiple threads and the refcounts are not atomic.

Reply:
I've fixed this with a force push to this branch.